src/Sets/Star.jl

```diff
@@ -0,0 +1,142 @@
+export Star,
+       center,
+       basis,
+       predicate
+
+"""
+    Star{N, VN<:AbstractVector{N}, MN<:AbstractMatrix{N}, PT<:AbstractPolyhedron{N}} <: AbstractStar{N}
+
+Type that represents a generalized star set where the predicate is polyhedral.
+
+### Fields
+
+- `c` -- vector that represents the center
+- `V` -- matrix where each column corresponds to a basis vector
+- `P` -- polyhedral set that represents the predicate
+
+### Notes
+
+The predicate function is implemented as a conjunction of linear constraints.
+This definition can be found in reference [1].
+
+### Examples
+
+This example is drawn from Example 1 in [1]. Consider the two-dimensional plane
+``\\mathbb{R}^2``. Let
+
+```jldoctest star_constructor
+julia> V = [[1.0, 0.0], [0.0, 1.0]]
```

Comment:
Here is another example, taken from Fig. 1 in "Star-based reachability analysis of DNNs"

```julia
c = zeros(2)
V = [1 0; 0 1.]
P = BallInf(zeros(2), 1.0)
S = Star(c, V, P)
```

Same example but instantiating with a polygon in constraint representation. Please note that `convert(HPolytope, BallInf(...))` will produce a polytope whose normal vectors are `SingleEntryVector` (SEV, ie. an efficient representation for vectors proportional to canonical basis vectors of R^n); the problem is that the new constraint obtained by intersection is not in general a SEV, hence we can't push onto the same array. OTOH `overapproximate(..., HPolygon)` creates constraints with `Vector{..}` normal vectors.

c = zeros(2)
V = [1 0; 0 1.]
P = overapproximate(BallInf(zeros(2), 1.0), HPolygon)
S = Star(c, V, P)
H = rand(HalfSpace)

intersection!(S, H) # in-place : add one more constraint to X

intersection(S, H)  # return a new star
```